test/Transactions/Orleans.Transactions.Tests/Runners/TransactionAttributionTestRunner.cs

```diff
@@ -0,0 +1,166 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Orleans.Transactions.Tests
+{
+    public abstract class TransactionAttributionTestRunner
+    {
+        protected readonly IGrainFactory grainFactory;
+        protected readonly ITestOutputHelper output;
+
+        protected TransactionAttributionTestRunner(IGrainFactory grainFactory, ITestOutputHelper output)
+        {
+            this.output = output;
+            this.grainFactory = grainFactory;
+        }
+
+        [Fact]
+        public async Task FanoutToAllAttributesTest()
+        {
+            ITransactionAttributionGrain root = this.grainFactory.GetTransactionAttributionGrain(Guid.NewGuid());
+            Dictionary<int, List<ITransactionAttributionGrain>> tiers = new Dictionary<int, List<ITransactionAttributionGrain>>
+            {
+                [1] = new List<ITransactionAttributionGrain>(new[] {
```

Comment:
Could you add a comment describing why we start at 1 here? After reading the code it becomes clear, but it's not obvious at first. Maybe something like `// Tier 0 is the root grain. We pass a collection of grains in tier 1 so that tier 0 will fan out to those grains.`

Reply:
I will simplify to use array.  Should be clearer... i hope :/